test/e2e/parametrisable_templates/parametrisable_templates_test.go

```diff
@@ -0,0 +1,440 @@
+package parametrisabletemplates
+
+import (
+	"fmt"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("Template variables in the device configuraion", func() {
+	var (
+		harness  *e2e.Harness
+		deviceId string
+	)
+
+	BeforeEach(func() {
+		harness = e2e.NewTestHarness()
+		deviceId = harness.StartVMAndEnroll()
+	})
+
+	AfterEach(func() {
+		harness.Cleanup(true)
+		err := harness.CleanUpAllResources()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("parametrisable_templates", func() {
+		It(`Verifies that Flightctl fleet resource supports parametrisable device
+		    templates to configure items that are specific to an individual device
+			or a group of devices selected by labels`, Label("75486"), func() {
+
+			By("Create a fleet with template variables in InlineConfigProviderSpec")
+			err := configProviderSpec.FromInlineConfigProviderSpec(inlineConfigValidWithFunction)
+			Expect(err).ToNot(HaveOccurred())
+			err = harness.CreateTestFleetWithConfig(fleetTestName, testFleetSelector, configProviderSpec)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check that the device status is Online")
+			CheckDeviceOnlineStatus(harness, deviceId)
+
+			By("Add the fleet selector and the team label to the device")
+			nextRenderedVersion, err := harness.PrepareNextDeviceVersion(deviceId)
+			Expect(err).ToNot(HaveOccurred())
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				device.Metadata.Labels = &map[string]string{
+					fleetSelectorKey: fleetSelectorValue,
+					teamLabelKey:     teamLabelValue,
+				}
+				logrus.Infof("Updating %s with label %s=%s and %s=%s", deviceId,
+					fleetSelectorKey, fleetSelectorValue, teamLabelKey, teamLabelValue)
+			})
+
+			By("Verify the Device is updated with the labels")
+			response, err := harness.Client.ReadDeviceWithResponse(harness.Context, deviceId)
+			Expect(err).ToNot(HaveOccurred())
+			device := response.JSON200
+			Expect(device).ToNot(BeNil(), "failed to read updated device")
+			responseSelectorLabelValue := (*device.Metadata.Labels)[fleetSelectorKey]
+			Expect(responseSelectorLabelValue).To(ContainSubstring(fleetSelectorValue))
+			responseTeamLabelValue := (*device.Metadata.Labels)[teamLabelKey]
+			Expect(responseTeamLabelValue).To(ContainSubstring(teamLabelValue))
+
+			By("Wait for the device to get the fleet configuration")
+			err = harness.WaitForDeviceNewRenderedVersion(deviceId, nextRenderedVersion)
```

Comment:
nice :)

Reply:
You need to account for how deterministic this will be (i.e. the renderedVersion could potentially change for other reasons over time as code changes?)